packages/editor/src/browser/editor-command.ts

```diff
@@ -242,9 +241,15 @@ export class EditorCommandContribution implements CommandContribution {
 
     @postConstruct()
     protected init(): void {
-        this.editorPreferences.onPreferenceChanged(e => {
-            if (e.preferenceName === 'files.autoSave' && e.newValue !== 'off') {
-                this.shell.saveAll();
+        this.preferencesService.ready.then(() => {
```

Comment:
Could be, but then `SaveResourceService` (from `@theia/core`) has an indirect dependency on `@theia/editor`, since the `files.autoSave` preference is defined in there. Are you in favor of that?